Prepare documentation page for the Update Manager configuration (#248)
eclipse-kanto/kanto: web/site/content/docs/references/updatemanager/update-manager-config.md

---
title: "Configuration"
type: docs
description: >
  Customize the update manager.
weight: 1
---

### Properties

To control all aspects of the update manager.

| Property | Type | Default | Description |
| - | - | - | - |
| **General** | | | |

> | domain | string | device | Specify the Domain of this update agent, used as MQTT topic prefix |

Reviewer: ```suggestion
| domain | string | device | The domain of this update agent, used as MQTT topic prefix |
```

> | domains | string | containers,self-update,safety-domain| Specify a comma-separated list of domains handled by the update manager |

Reviewer: ```suggestion
| domains | string | containers,self-update,safety-domain| A comma-separated list of domains handled by the update manager |

```

> | phaseTimeout | string | 10m | Specify the timeout for completing an Update Orchestration phase |

Reviewer: ```suggestion
| phaseTimeout | string | 10m | Timeout as duration string for completing an Update Orchestration phase |
```

> | rebootAfter | string | 30s | Specify the timeout in cron format to wait before a reboot process is initiated after successful update operation |

Reviewer: ```suggestion
| rebootAfter | string | 30s | Time period in cron format to wait before a reboot process is initiated after successful update operation |

```

> | rebootEnabled | bool | true | Specify a flag that controls the enabling/disabling of the reboot process after successful update operation |

Reviewer: ```suggestion
| rebootEnabled | bool | true | Enable the reboot process after successful update operation |
```

> | reportFeedbackInterval | string | 1m | Specify the time interval for reporting intermediate desired state feedback messages during an active update operation |

Reviewer: ```suggestion
| reportFeedbackInterval | string | 1m | Time interval as duration string for reporting intermediate desired state feedback messages during an active update operation |
```

> | currentStateDelay | string | 30s | Specify the time delay for reporting current state messages |

Reviewer: Adapt all other time properties in a similar way

> | **Domain agent** | | | |

Reviewer: I guess it should match the json template.
```suggestion
| **Domain agents** | | | |
```

| readTimeout | int | 1m | Specify the read timeout for the given domain |

> | rebootRequired | bool | false | Specify the reboot required flag for the given domain |

Reviewer: `name` is missing, but is listed in the json template

> | rebootRequired | bool | false | Specify the reboot required flag for the given domain |

Reviewer: ```suggestion
| rebootRequired | bool | false | Require a reboot for the given domain |
```

> | **Local connectivity** | | | |

Reviewer: Please check the `Digital Twin - connectivity` section [here](https://eclipse.dev/kanto/docs/references/containers/container-manager-config/) and copy the descriptions from there.

| brokerUrl | string | tcp://localhost:1883 | Specify the MQTT broker URL to connect to |
| keepAlive | bool | 20000 | Specify the keep alive duration for the MQTT requests in milliseconds |
| acknowledgeTimeout | int | 15000 | Specify the acknowledgement timeout for the MQTT requests in milliseconds |
| clientUsername | string | | Specify the MQTT client username to authenticate with |
| clientPassword | string | | Specify the MQTT client password to authenticate with |
| connectTimeout | int | 30000 | Specify the connect timeout for the MQTT in milliseconds |
| disconnectTimeout | int | 250 | Specify the disconnection timeout for the MQTT connection in milliseconds |
| subscribeTimeout | int | 15000 | Specify the subscribe timeout for the MQTT requests in milliseconds |
| unsubscribeTimeout | int | 5000 | Specify the unsubscribe timeout for the MQTT requests in milliseconds |

> | **Logging** | | | |

Reviewer: Please check the `Logging` section [here](https://eclipse.dev/kanto/docs/references/containers/container-manager-config/) and copy the descriptions from there.

| logFile | string | | Set the log file |
| logLevel | string | INFO | Set the log level - possible values are ERROR, WARN, INFO, DEBUG, TRACE |
| logFileCount | int | 5 | Set the maximum number of old log files to retain |
| logFileMaxAge | int | 28 | Set the maximum number of days to retain old log files based on the timestamp encoded in their filename |
| logFileSize | int | 2 | Set the maximum size in megabytes of the log file before it gets rotated |

> ### Template

Reviewer: Example section is missing, please check the other similar reference docs and add one.

The configuration can be further adjusted according to the use case.
The following template illustrates all possible properties with their default values.

```json
{
	"domain": "device",
	"domains": "containers,self-update,safety-domain",
	"agents": {
		"containers": {
			"name": "containers",
			"rebootRequired": false,
			"readTimeout": "30s"
		},
		"self-update": {
			"name": "self-update",
			"rebootRequired": true,
			"readTimeout": "30s"
		},
		"safety-domain": {
			"name": "safety-domain",
			"rebootRequired": true,
			"readTimeout": "30s"
		}
	},
	"log": {
		"logFile": "",
		"logLevel": "DEBUG",
		"logFileCount": 5,
		"logFileMaxAge": 28,
		"logFileSize": 2
	},

> 	"mqtt": {

Reviewer: IMO `mqtt` should be `connection`,  `local_connection`, `connectivity` or `local_connectivity` or something similar. @stoyan-zoubev could you fix it in the update-manager, please? Sorry, I have missed that

Reviewer 2: Yes, we shall align the configuration properties for local connectivity with the other Kanto components.
